libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationEngineFactory.kt

```diff
@@ -0,0 +1,73 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import com.mapbox.services.android.navigation.v5.navigation.camera.Camera
+import com.mapbox.services.android.navigation.v5.navigation.camera.SimpleCamera
+import com.mapbox.services.android.navigation.v5.offroute.OffRoute
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteDetector
+import com.mapbox.services.android.navigation.v5.route.FasterRoute
+import com.mapbox.services.android.navigation.v5.route.FasterRouteDetector
+import com.mapbox.services.android.navigation.v5.snap.Snap
+import com.mapbox.services.android.navigation.v5.snap.SnapToRoute
+
+class NavigationEngineFactory {
+
+    private var offRouteEngine: OffRoute? = null
+    private var fasterRouteEngine: FasterRoute? = null
+    private var snapEngine: Snap? = null
+    private var cameraEngine: Camera? = null
+
+    init {
+        initializeDefaultEngines()
```

Comment:
Could we move initialization to properties definition and make types non-nullable? Methods from `MapboxNavigation` are tagged as `@NonNull` e.g. https://github.com/mapbox/mapbox-navigation-android/blob/53171c7c38874fa9f840261118af0e2dc3570f62/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java#L696-L697 https://github.com/mapbox/mapbox-navigation-android/blob/53171c7c38874fa9f840261118af0e2dc3570f62/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java#L682

Reply:
👍